sklearn/impute/tests/test_impute.py

```diff
@@ -971,6 +971,58 @@ def test_iterative_imputer_catch_warning():
     assert not np.any(np.isnan(X_fill))
 
 
+@pytest.mark.parametrize("min_value, max_value, correct_output",
+                         [(0,  # scalars cast to the correct arrays?
+                           100,
+                           np.array([[0] * 3, [100] * 3])),
+                          ([0, 0, 0],  # vectors cast to the same
+                           [100, 100, 100],  # correct arrays?
+                           np.array([[0] * 3, [100] * 3])),
+                          (None,  # None cast to vector of inf?
+                           None,
+                           np.array([[-np.inf] * 3, [np.inf] * 3])),
+                          (-np.inf,  # inf cast to vector of inf?
+                           np.inf,
+                           np.array([[-np.inf] * 3, [np.inf] * 3])),
+                          ([-5, 5, 10],  # vectors processed properly?
+                           [100, 200, 300],
+                           np.array([[-5, 5, 10], [100, 200, 300]])),
+                          ([-5, None, 10],  # vectors of numbers+None
+                           [100, 200, None],  # cast correctly?
+                           np.array([[-5, -np.inf, 10], [100, 200, np.inf]]))])
+def test_iterative_imputer_min_max_array_like(min_value,
+                                              max_value,
+                                              correct_output):
+    # check that passing scalar or array-like
+    # for min_value and max_value in IterativeImputer works
+    X = np.random.random((10, 3))
+    imputer = IterativeImputer(min_value=min_value, max_value=max_value)
+    imputer.fit(X)
+
+    assert (isinstance(imputer._min_value, np.ndarray) and
+            isinstance(imputer._max_value, np.ndarray))
+    assert ((imputer._min_value.shape[0] == X.shape[1]) and
+            (imputer._max_value.shape[0] == X.shape[1]))
+
+    assert_array_equal(correct_output[0, :], imputer._min_value)
+    assert_array_equal(correct_output[1, :], imputer._max_value)
+
+
+@pytest.mark.parametrize("min_value, max_value",
+                         [(100, 0), (np.inf, -np.inf),
+                          ([-5, 5], [100, 200, 0])])
+def test_iterative_imputer_catch_min_max_error(min_value, max_value):
+    # check that passing scalar or array-like
+    # for min_value and max_value in IterativeImputer works
```

Comment:
We don't need comments as well. For instance here we cannot know the the case that you are testing in practise because you are not matching the error message.

Thus if you are doing `pytest.raises(ValueError, match="err_msg")`, you can pass the expected (partial error messgae). `min_value should be of shape` or `One (or more) ...`. Then the test will be self explenatory.

Reply:
Done!